apps/console/app/mocks/branches.ts

```
@@ -31,11 +33,63 @@ export const branchHandlers = [
 
   http.get<{ agentSlug: string }>(
     "/api/v1/agents/:agentSlug/branches",
-    async () => {
-      const branches = db.branch.getAll();
+    async ({ params }) => {
+      try {
+        const branches = db.branch.findMany({
+          where: { agentSlug: { equals: params.agentSlug } },
+        });
+
+        await delay(1000);
+        return HttpResponse.json(branches);
+      } catch {
+        return new HttpResponse("Failed to fetch branches", { status: 500 });
+      }
```

Why do you try/catch to still return a 500? Do you think this is necessary? Why not just return the original error?

---

I will return the original error like this 
```ts
return new HttpResponse(error instanceof Error ? error.message : String(error), { status: 500 },);
```